lib/stockpot/errors.rb

```
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Errors
+  def self.rescue_error(error)
+    logger = Logger.new(STDERR)
+    logger.warn(error)
+
+    case error
+    when NameError
+      return return_error(error.to_s, :bad_request)
+    when PG::Error
+      return return_error("Database error: #{error}", :server_error)
+    else
+      return return_error(error.to_s, :server_error)
+    end
+  end
```

Can we also bring in validation errors here? 
Here is a reference: https://guides.rubyonrails.org/active_record_validations.html

---

An example would be trying to create the same shippingOption twice.

---

I have not tested this but something like this might work 
```
when ActiveModel::ValidationError => invalid
  ## invalid.model.errors -> You could have access to all model errors this way
end
```

---

They probably would have been caught by the else, but I've added in their own case. :)

---

Yeah, let's return the specific errors not the whole trace....I find no benefit on returning the whole backtrace in JSON

---

It actually doesn't, but still changed it to be `error.to_s` instead. :)

---

Example:

```json
{
    "error": {
        "status": "server_error",
        "message": "ActiveRecord error: Validation failed: Target zip has already been taken"
    }
}
```

---

That looks very nice!! :)